app/models/on_duty_locator.rb

```diff
@@ -0,0 +1,32 @@
+class OnDutyLocator
+  def initialize(time, rota_slots)
+    @time = time
+    @rota_slots = rota_slots
+  end
+
+  def locate
+    on_duty_organisation_uid
+  end
+
+  private
+
+  attr_reader :time, :rota_slots
+
+  def on_duty_organisation_uid
+    slots_with_on_duty_shift.map(&:organisation_uid)
+  end
+
+  def slots_with_on_duty_shift
+    applicable_slot = find_latest_slot
+
+    if applicable_slot.nil? || (applicable_slot.ending_time.present? && applicable_slot.ending_time < time)
+      []
+    else
+      rota_slots.where(starting_time: applicable_slot.starting_time, shift_id: applicable_slot.shift_id)
```

Comment:
This will cause another round trip to the DB. `rota_slots` is an ActiveRecordRelation already, could we try traversing the set without another query?